src/wallet/wallet.cpp

```diff
@@ -1507,6 +1507,34 @@ bool DummySignInput(const SigningProvider& provider, CTxIn &tx_in, const CTxOut
     return true;
 }
 
+void FillInputToWeight(CTxIn& txin, int64_t target_weight)
```

Comment:
Apparently `GetTransactionInputWeight` includes the empty witness, and I forgot to include the empty scriptSig in my calculation.

I've changed restriction to be at least 165 bytes. `FillInputToWeight` will also check the size meets the minimum. If it does not, it will just do nothing rather than crashing. There are also unit tests for this lower bound.